src/content/docs/apm/agents/go-agent/instrumentation/go-agent-code-level-metrics-instrument.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: Go agent code-level metrics instrumentation
+tags:
+  - Agents
+  - Go agent
+  - Instrumentation
+metaDescription: Add source code context to your application traces, associating the trace data with the specific code being instrumented.
+---
+
+After you've turned on code-level metrics using the steps in [Go agent code-level metrics configuration](content/docs/apm/agents/go-agent/configuration/go-agent-code-level-metrics-config), you can fine-tune how your metrics are collected by using additional instrumentation. For general information about code-level metrics, see [Performance monitoring with CodeStream](/docs/codestream/how-use-codestream/performance-monitoring#code-level).
+
+## Instrumenting Transactions [#instrument-transactions-clm]
+
+With code-level metrics enabled, the Go Agent will add source code context information to any transaction started by a call
+to the `StartTransaction` method of the application. Without changing any of your existing instrumentation code, this means
+that the source code location will be added based on the function which called `StartTransaction`. This is true even if
+`StartTransaction` is called for you by an instrumentation package.
+
+However, if you wish to exert some manual control over how code-level metrics are collected for a particular transaction,
+you may add a number of `newrelic.TraceOption` functions as described below:
+
```

Comment:
How would you feel about making all the ### headings below collapsers? They all seem to relate to the same concern, issue, and I think it would make it more clear that they're part of a group. It would also contribute to a more seamless presentation.

Reply:
@homelessbirds, I'm hesitant to add collapsers under the Transactions heading because we'll have a wall of collapsers. Are you thinking that the smaller h3 fonts are not a big enough contrast to the h2? I'm also hoping users see the h2s in the right navigation to keep them oriented.